### DISCUSSION TITLE
How to create a new btn component from existing btn in tailwind config ?

### DISCUSSION THREAD

**Discussion by geminigeek** [saadeghi/daisyui#1660] [Q&A] [answered]
hi,

i want to customize btn like below, its working if i put this in my main layout file, but this is creating a new css file in final build, how can i set this up in tailwind.config.js , so these styles/css remain in one css file in final production ? 
i am using astro.

```
.tag-link {
        @apply btn btn-outline w-[170px] sm:w-[180px] text-[14px] capitalize text-gray-600 dark:text-gray-400 hover:dark:bg-orange-300 hover:dark:text-black;
      }
```

**Comment by saadeghi** [comment, marked as answer]
The easiest way would be putting this style in your the CSS file
```css
@tailwind base;
@tailwind components;
@tailwind utilities;

/* Here */
```

**Reply by geminigeek** [reply]
Hi, thanks for your reply, I know it can be done like this, but there is no  css file in astro tailwind integration?  That's why I asked specifically how to do it in tailwind config !, May be I am not aware, I am new to frontend.

**Reply by saadeghi** [reply]
Using Astor's Tailwind integration, you can add your own CSS file this way: https://docs.astro.build/en/guides/integrations-guide/tailwind/#configapplybasestyles

You can add styles inside Tailwind config using Tailwind's plugin API but it's easier to add it in the CSS file instead.

Let me know if you have any questions

**Reply by geminigeek** [reply]
hi saadeghi,

thanks for your help, it worked :)

**Comment by geminigeek** [comment]
hi,

it seems astro has changes the was they use "applyBaseStyles" , now the above solution do not work ! , can you provide a solution for this here is the relevant info on the docs https://docs.astro.build/en/guides/integrations-guide/tailwind/#troubleshooting

error i am getting is 

```
The sm: class does not exist. If sm: is a custom class, make sure it is defined within a @layer directive.
```

this is my base.css

```
@tailwind base;
@tailwind components;
@tailwind utilities;

.tag-link {
  @apply btn btn-outline w-[170px] sm: w-[180px] text-[14px] capitalize text-gray-600 dark:text-gray-400 hover:dark:bg-orange-300 hover:dark:text-black;
}

```

**Reply by geminigeek** [reply]
it seems my code got  some spaces in class names let me check and get back , just realized seeing it here

**Reply by geminigeek** [reply]
my bad i had space in class name , sorry to bother
